switchapi/es2k/lnw_v3/switch_pd_p4_name_routing.h

```diff
@@ -0,0 +1,176 @@
+
+#define LNW_ECMP_NEXTHOP_TABLE_ACTION_SET_ECMP_NEXTHOP_INFO_DMAC \
+  "linux_networking_control.ecmp_set_nexthop_info_dmac"
+#define LNW_ACTION_SET_ECMP_NEXTHOP_PARAM_RIF "router_interface_id"
+#define LNW_ACTION_SET_ECMP_NEXTHOP_PARAM_DMAC_HIGH "dmac_high"
+#define LNW_ACTION_SET_ECMP_NEXTHOP_PARAM_DMAC_LOW "dmac_low"
+#define LNW_ACTION_SET_ECMP_NEXTHOP_PARAM_EGRESS_PORT "egress_port"
+
+/* ECMP_HASH_TABLE */
+#define LNW_ECMP_HASH_TABLE "linux_networking_control.ecmp_hash_table"
+
+#define LNW_ECMP_HASH_TABLE_KEY_HOST_INFO_TX_EXT_FLEX \
+  "user_meta.cmeta.flex[15:0]"
+#define LNW_ECMP_HASH_TABLE_KEY_META_COMMON_HASH "vmeta.common.hash[2:0]"
+#define LNW_ECMP_HASH_TABLE_KEY_ZERO_PADDING "zero_padding"
+
+#define LNW_ECMP_HASH_TABLE_ACTION_SET_NEXTHOP_ID \
+  "linux_networking_control.set_nexthop_id"
+
+#define LNW_ECMP_HASH_SIZE 65536
+
+/* Only 3 bits is allocated for hash size per group in LNW.p4
+ * check LNW_ECMP_HASH_TABLE_KEY_META_COMMON_HASH */
+#define LNW_ECMP_PER_GROUP_HASH_SIZE 8
+
+/* TX_LAG_TABLE */
+#define LNW_TX_LAG_TABLE "linux_networking_control.tx_lag_table"
+
+#define LNW_TX_LAG_TABLE_KEY_LAG_ID "user_meta.cmeta.lag_group_id"
+#define LNW_TX_LAG_TABLE_KEY_VMETA_COMMON_HASH "hash"
+
+#define LNW_TX_LAG_TABLE_ACTION_SET_EGRESS_PORT \
+  "linux_networking_control.set_egress_port"
+
+#define ACTION_SET_EGRESS_PORT_PARAM_EGRESS_PORT "egress_port"
+
+#define LNW_LAG_HASH_SIZE 65536
+
+/* Only 3 bits is allocated for hash size per group in LNW.p4
+ * check LNW_TX_LAG_TABLE_KEY_VMETA_COMMON_HASH */
+#define LNW_LAG_PER_GROUP_HASH_SIZE 8
+
+/* IPV4_TABLE */
+#define LNW_IPV4_TABLE "linux_networking_control.ipv4_table"
+
+// TODO_LNW_MEV: One additional key for MEV(ipv4_table has 2 keys for MEV)
+#define LNW_IPV4_TABLE_KEY_IPV4_TABLE_LPM_ROOT "ipv4_table_lpm_root"
+
+#define LNW_IPV4_TABLE_KEY_IPV4_DST_MATCH "ipv4_dst_match"
+
+#define LNW_IPV4_TABLE_ACTION_IPV4_SET_NEXTHOP_ID \
+  "linux_networking_control.ipv4_set_nexthop_id"
+#define LNW_ACTION_SET_NEXTHOP_ID_PARAM_NEXTHOP_ID "nexthop_id"
+
+#define LNW_IPV4_TABLE_ACTION_IPV4_SET_NEXTHOP_ID_WITH_MIRROR \
+  "linux_networking_control.ipv4_set_nexthop_id_with_mirror"
+
+#define LNW_IPV4_TABLE_ACTION_ECMP_HASH_ACTION \
+  "linux_networking_control.ecmp_hash_action"
+#define LNW_ACTION_ECMP_HASH_ACTION_PARAM_ECMP_GROUP_ID "ecmp_group_id"
+
+/* IPV6_TABLE */
+#define LNW_IPV6_TABLE "linux_networking_control.ipv6_table"
+
+// TODO_LNW_MEV: One additional key for MEV(ipv6_table has 2 keys for MEV)
+#define LNW_IPV6_TABLE_KEY_IPV6_TABLE_LPM_ROOT "ipv6_table_lpm_root"
+
+#define LNW_IPV6_TABLE_KEY_IPV6_DST_MATCH "ipv6_dst_match"
+
+#define LNW_IPV6_TABLE_ACTION_IPV6_SET_NEXTHOP_ID \
+  "linux_networking_control.ipv6_set_nexthop_id"
+
+#define LNW_IPV6_TABLE_ACTION_IPV6_SET_NEXTHOP_ID_WITH_MIRROR \
+  "linux_networking_control.ipv6_set_nexthop_id_with_mirror"
+
+#define LNW_IPV6_TABLE_ACTION_ECMP_V6_HASH_ACTION \
+  "linux_networking_control.ecmp_v6_hash_action"
+
+#ifdef __cplusplus
+}
```

Comment:
1. Should be removed, along with the superfluous `extern "C"`.
2. I generally include an end-of-line comment on the closing brace (`// extern "C"`), to make it clear what kind of block we're closing. I do something similar for namespaces in C++.

This is another change I've been making as I edit files.